File: src/main/java/com/aladdinworks9/service/CapacityReportService.java
package com.aladdinworks9.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks9.domain.CapacityReport;
import com.aladdinworks9.dto.CapacityReportDTO;
import com.aladdinworks9.dto.CapacityReportSearchDTO;
import com.aladdinworks9.dto.CapacityReportPageDTO;
import com.aladdinworks9.dto.CapacityReportConvertCriteriaDTO;
import com.aladdinworks9.service.GenericService;
import com.aladdinworks9.dto.common.RequestDTO;
import com.aladdinworks9.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CapacityReportService extends GenericService<CapacityReport, Integer> {

	List<CapacityReport> findAll();

	ResultDTO addCapacityReport(CapacityReportDTO capacityReportDTO, RequestDTO requestDTO);

	ResultDTO updateCapacityReport(CapacityReportDTO capacityReportDTO, RequestDTO requestDTO);

    Page<CapacityReport> getAllCapacityReports(Pageable pageable);

    Page<CapacityReport> getAllCapacityReports(Specification<CapacityReport> spec, Pageable pageable);

	ResponseEntity<CapacityReportPageDTO> getCapacityReports(CapacityReportSearchDTO capacityReportSearchDTO);
	
	List<CapacityReportDTO> convertCapacityReportsToCapacityReportDTOs(List<CapacityReport> capacityReports, CapacityReportConvertCriteriaDTO convertCriteria);

	CapacityReportDTO getCapacityReportDTOById(Integer capacityReportId);







}

File: aladdinworks9_ui/src/services/CapacityReportService.js
import http from "../http-common"; 

class CapacityReportService {
  getAllCapacityReports(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/capacityReport/capacityReports`, searchDTO);
  }

  get(capacityReportId) {
    return this.getRequest(`/capacityReport/${capacityReportId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/capacityReport?field=${matchData}`, null);
  }

  addCapacityReport(data) {
    return http.post("/capacityReport/addCapacityReport", data);
  }

  update(data) {
  	return http.post("/capacityReport/updateCapacityReport", data);
  }
  
  uploadImage(data,capacityReportId) {
  	return http.postForm("/capacityReport/uploadImage/"+capacityReportId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CapacityReportService();
